docs/getting-started.md

```diff
@@ -0,0 +1,117 @@
+# Getting Started
+
+This document describes the anatomy of a common COS application and the language-agnostic steps
+to get started. See the sample applications for language-specific examples.
+
+## Generating the protos
+
+chief-of-state defines its public interfaces as [gRPC](https://grpc.io/) services
+and objects as [protobufs](https://developers.google.com/protocol-buffers) in
+`.proto` files. You will need to generate these for your application language
+to interact with chief-of-state and also to implement the requried write-handler
+and read-handler methods (more on those later).
+
+The protos are located in a github repo: https://github.com/namely/chief-of-state-protos
+
+A popular way of incorporating them into your project is via git submodules,
+which you can add like so:
+```sh
+# adds COS protos to proto/cos
+git submodule add git@github.com:namely/chief-of-state-protos ./protos/cos
+```
+
+From there, you can generate the COS interfaces for your language (see the
+official quick start guide for your language https://grpc.io/docs/languages/).
+
+As an example, one would use [protoc](https://grpc.io/docs/protoc-installation/)
+to generate java source code like so:
+```sh
+export SRC_DIR=protos/cos
+export DST_DIR=build/gen
+# generate everything in protos/cos/chief_of_state/v1/common.proto
+protoc -I=$SRC_DIR --java_out=$DST_DIR $SRC_DIR/foo.proto
+# generate ALL .proto files in protos/cos
+find $SRC_DIR -name "*.proto" | xargs -I{} protoc -I=$SRC_DIR --java_out=$DST_DIR {}
+```
+
+**Pro tip:** many languages have wrappers around protoc to make this easier,
+like [scalap-pb](https://scalapb.github.io/) for scala (which chief-of-state
+uses internally). Namely also offers an open-source docker-based solution
+[namely/docker-protoc](https://github.com/namely/docker-protoc) with support
+for many languages.
+
+## Implement a "Write Side"
+
+Drawing from CQRS terminology, COS defines a [writeside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/writeside.proto) that you must implement to define how state
+is created and updated.
+
+### Methods
+`HandleCommand` accepts a command (think "request") and a prior state and returns an event.
+For example, given a command to `UpdateUserEmail` and a User, this RPC might return
+an event called `UserEmailUpdated`. This method is allowed to throw exceptions
+(as gRPC error statuses) if the inbound command (request) is invalid.
+
+`HandleEvent` accepts an event and the prior state of an entity and returns a new state.
+For example, given a UserEmailUpdated event and a User, this RPC would return a
+new User instance with the email updated. It's advisable to do all your
+validations in `HandleCommand` such that this method is unlikely to encounter
+invalid data. If `HandleCommand` fails, `HandleEvent` is not called for that
+command.
+
+### Registration
```

Comment:
maybe we have this explained somewhere else but I would consider reiterate what is explained in the diagram and then how cos requires some mandatory vars with the write handler vars among them.

Reply:
A few lines down there's a link to the full config doc.

Regarding the diagram, yeah maybe that and a few sentences describing the components at the top. I'm torn about the scope of this doc.